src/libOpenImageIO/imagebufalgo_xform.cpp

```diff
@@ -307,13 +408,92 @@ warp_impl(ImageBuf& dst, const ImageBuf& src, const Imath::M33f& M,
+// Is it in the `obsolete` list? If so, return false.
+// Is it in neither? Return false.
+static bool
+IBA_find_optional(const ParamValue& option, cspan<ustring> recognized,
+                  cspan<ustring> obsolete = {})
+{
+    for (auto&& r : recognized)
+        if (option.name() == r)
+            return true;
+    for (auto&& o : obsolete)
+        if (option.name() == o)
+            return false;
+    return false;
+}
+
+
+
+// Given list of `options`, check if any are unrecognized or obsolete.
+// Return true if all are fine, false if not.
+static bool
+IBA_check_optional(ImageBufAlgo::KWArgs options, cspan<ustring> recognized,
+                   cspan<ustring> obsolete = {})
+{
+    bool ok = true;
+    for (auto&& pv : options) {
+        ok &= IBA_find_optional(pv, recognized, obsolete);
+    }
+    return ok;
+}
+
+
+
+bool
+ImageBufAlgo::warp(ImageBuf& dst, const ImageBuf& src, M33fParam M,
+                   KWArgs options, ROI roi, int nthreads)
+{
+    static ustring recognized[] = { "filtername"_us,    "filterwidth"_us,
+                                    "wrap"_us,          "edgeclamp"_us,
+                                    "recompute_roi"_us, "filterptr"_us };
+    IBA_check_optional(options, recognized);
+
+    Filter2D::ref filterref;
+    if (options.contains("filterptr", TypePointer)) {
+        auto f = options.find("filterptr", TypePointer);
+        OIIO_DASSERT(f != options.cend());
+        filterref = Filter2D::ref(f->get<const Filter2D*>(),
+                                  Filter2D::no_destroy);
+    } else {
+        filterref = get_warp_filter(options.get_string("filtername"),
+                                    options.get_float("filterwidth"), dst);
+        if (!filterref)
+            return false;  // error issued in get_warp_filter
+    }
+    if (!filterref) {
+        dst.errorfmt("Invalid filter");
+        return false;
+    }
+
+    ImageBuf::WrapMode wrap = ImageBuf::WrapMode::WrapDefault;
+    auto wrapparam          = options.find("wrap");
+    if (wrapparam != options.end()) {
+        if (wrapparam->type() == TypeString)
+            wrap = ImageBuf::WrapMode_from_string(wrapparam->get_ustring());
+        else
+            wrap = (ImageBuf::WrapMode)wrapparam->get_int();
+    }
+    bool recompute_roi = options.get_int("recompute_roi", 0);
+    bool edgeclamp     = options.get_int("edgeclamp", 0);
+
+    return warp_impl(dst, src, M, filterref.get(), recompute_roi, wrap,
+                     edgeclamp, roi, nthreads);
+}
+
+
+
 bool
 ImageBufAlgo::warp(ImageBuf& dst, const ImageBuf& src, M33fParam M,
                    const Filter2D* filter, bool recompute_roi,
                    ImageBuf::WrapMode wrap, ROI roi, int nthreads)
 {
-    return warp_impl(dst, src, M, filter, recompute_roi, wrap,
-                     false /*edgeclamp*/, roi, nthreads);
+    return warp(dst, src, M,
+                { { "filterptr", TypePointer, 1, &filter },
```

Comment:
What is the `1`? I can go look for the corresponding code to find out, but this is harder to follow than the old direct passing of a pointer. Is there something we can do to make this easier to read? If this is a count, could we add some logic so not including the count it defaults to 1? 

In OIIO internals, it's fine if we require the reader to be an advanced user of OIIO, but what about when users of OIIO need to pass optional pointers? We should try to make that not too painful.

Reply:
Ah, this is related to the actual constructor for a ParmValue, which takes 
* name
* the type T
* the number of T items
* a `T*` pointing to the values to copy

This is the general constructor, but it's a little awkward for just one item of common types, so we have helper constructors for the extremely common cases of needing one int, float, or string. There is currently no direct simple constructor for pointer types, but maybe I should add one. I had noticed when I wrote this that it was falling into the awkward general case. Let me see what I can throw together.

Reply:
The update I pushed helps this, I think.